user/views.py

```diff
@@ -1,17 +1,55 @@
 from django.shortcuts import render, redirect
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth import authenticate, login, logout
+from django.contrib import messages
+from .models import User
 
-# Create your views here.
 
 
 
+def home(request):
+    if request.user.is_authenticated:
+        return render(request,'tweet/home.html')
+    return render(request,'tweet/login.html')
 
-def check_login(request):
+def signup(request):
     if request.user.is_authenticated:
         return redirect('home')
-    return render(request,'tweet/index.html')
+    if request.method == 'GET':
+        return render(request,'tweet/register.html')
+    email = request.POST.get('email')
+    first_name = request.POST.get('first_name')
+    last_name = request.POST.get('last_name')
+    password = make_password(request.POST.get('password'))
+    try:
+        User.objects.create(username=email, password=password, first_name=first_name, last_name=last_name, email=email)
+        messages.success(request, "Login to continue")
+    except Exception as e:
+        messages.error(request, e)
+    return redirect('home')
 
+def signin(request):
+    if request.user.is_authenticated:
+        return redirect('home')
```

Comment:
can we not add this as separate function and add as a before callback

Reply:
you are saying to make a separate function for this and where ever we need to check user is authenticated we can just call the function. If I'm getting it right we can do that also.

Reply:
I tried to do so and made a function check_login
```python
def check_login(request):
    if request.user.is_authenticated:
        return True
    return False
``` 

but this will just add some line of code as whenever we need to check this we will call the function with a condition that return true the go to homepage otherwise execute the rest of the code

Reply:
you can put redirect in this function

Reply:
```python
def check_login(request):
    if request.user.is_authenticated:
        return redirect('home')
``` 
```python
def check_login(request):
    if request.user.is_authenticated:
        redirect('home')
```

I've checked these don't work